### ISSUE TITLE
Gabby has no elixirs to sell

### ISSUE THREAD

**Issue by raevenrises** [ja2-stracciatella/ja2-stracciatella#1961]
Gabby has no elixirs to sell. He only has empty jars to sell. This is my first time visiting him, after the crepitus have taken over a mine.

Windows 11 OS

| JA2S version | Game version        | Vanilla bug? | Used to work? |
| 0.21 | JA2 Gold 1.12 | Unknown, dont think so | Have not tried before |

[2024-03-10t23-16-31z-gabby.sav.txt](https://github.com/ja2-stracciatella/ja2-stracciatella/files/14552092/2024-03-10t23-16-31z-gabby.sav.txt)

I added a .txt file extension to the save game file so that Github would accept it.

**Comment by lynxlynxlynx** [comment]
Maybe the change is on a timer; it's unclear if this is a bug yet.

**Comment by momoko-h** [comment]
Yes, if I load that save and look at gArmsDealersInventory [4][218] (4: Gabby, 218: JAR_ELIXIR) I get the following: 

> $5 = {ubTotalItems = 0 '\000', ubPerfectItems = 0 '\000', ubStrayAmmo = 0 '\000', 
>   SpecialItem = std::vector of length 0, capacity 0, uiOrderArrivalTime = 42, ubQtyOnOrder = 2 '\002', 
>   fPreviouslyEligible = 1 '\001'}

Which means he doesn't currently have any but should get 2 elixirs on day 42.

However, I wonder why fPreviouslyEligible is 1; if this is really the first interaction with Gabby's store this should be 0, which would make the initial 3 elixirs available.

**Comment by raevenrises** [comment]
> Yes, if I load that save and look at gArmsDealersInventory [4][218] (4: Gabby, 218: JAR_ELIXIR) I get the following:
> 
> > $5 = {ubTotalItems = 0 '\000', ubPerfectItems = 0 '\000', ubStrayAmmo = 0 '\000',
> > SpecialItem = std::vector of length 0, capacity 0, uiOrderArrivalTime = 42, ubQtyOnOrder = 2 '\002',
> > fPreviouslyEligible = 1 '\001'}
> 
> Which means he doesn't currently have any but should get 2 elixirs on day 42.
> 
> However, I wonder why fPreviouslyEligible is 1; if this is really the first interaction with Gabby's store this should be 0, which would make the initial 3 elixirs available.

It is definitely the first interaction with him.
